instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -250,28 +254,22 @@ def instrument_app(
             app._before_request = _before_request
             app.before_request(_before_request)
             app.teardown_request(_teardown_request)
-            app._is_instrumented = True
+            app._is_instrumented_by_opentelemetry = True
         else:
             _logger.warning(
                 "Attempting to instrument Flask app while already instrumented"
             )
 
-    def _uninstrument(self, **kwargs):
-        flask.Flask = self._original_flask
-
-    def uninstrument_app(self, app):  # pylint: disable=no-self-use
-        if not hasattr(app, "_is_instrumented"):
-            app._is_instrumented = False
-
-        if app._is_instrumented:
+    @staticmethod
+    def uninstrument_app(app):
+        if hasattr(app, "_original_wsgi_app"):
             app.wsgi_app = app._original_wsgi_app
 
             # FIXME add support for other Flask blueprints that are not None
             app.before_request_funcs[None].remove(app._before_request)
             app.teardown_request_funcs[None].remove(_teardown_request)
             del app._original_wsgi_app
-
-            app._is_instrumented = False
+            app._is_instrumented_by_opentelemetry = False
```

Comment:
Should we rather delete this attribute when uninstrumented?

Reply:
Not a strong opinion on this. The field is really only used by OpenTelemetry.